rezerwacje-front/src/views/Exams/components/ExamsContent/ExamFormDialog.jsx

```diff
@@ -17,150 +16,302 @@ import {
+      function (e) {
+        alert("Wystąpił błąd.");
+      }
+    );
+  };
+
+  const handleSubmit = async event => {
+    event.preventDefault();
+    let room = -1;
+    if (roomId != -1)
+      room = getRoom();
+    setErrors(validateExamForm(courseId, roomId, getDate(selectedDate), getTime(startTime), getTime(endTime), periodDetails, room));
+    setIsSubmitting(true);
+  };
+
+  const handleCourseChange = event => setCourseId(event.target.value);
+
+  const handleRoomChange = event => setRoomId(event.target.value);
+
+  const handleStartTimeChange = (selectedTime) => setStartTime(selectedTime);
+
+  const handleEndTimeChange = (selectedTime) => setEndTime(selectedTime);
+
+  const createRoomData = room => {
+    return {
+      "id": room.id,
+      "room": room.room,
+      "availableFrom": room.availableFrom,
+      "availableTo": room.availableTo
+    }
+  };
+
+  const getRooms = rooms => rooms.map(room => createRoomData(room));
+
+  const getTime = time => time.toLocaleString('sv', { timeZoneName: 'short' }).split(" ")[1].slice(0, 6) + "00";
+
+  const getDate = date => date.toISOString().split("T")[0].slice(0, 19);
+
+  const getRoom = () => rooms.filter(room => room.id == roomId)[0];
+
+  useEffect(() => {
+    if (Object.keys(errors).length === 0 && isSubmitting) {
+      submit();
+    } else {
+      console.log(errors);
+    }
+  }, [errors]);
+
+  useEffect(() => {
+    if (props.recruitment === "") return;
+    fetch(`/api/recruitment/${props.recruitment}`)
+      .then(res => res.json())
+      .then(json => {
+        setRecruitmentName(json["recruitment"]["name"]);
+      })
+      .catch(e => console.log(e));
+  }, [props.recruitment]);
+
+  useEffect(() => {
+    if (props.period === "") return;
+    fetch(`/api/recruitment-period/${props.period}`)
+      .then(res => res.json())
+      .then(json => {
+        const periodJson = json["recruitmentPeriod"];
+        const start = periodJson["startDate"];
+        const end = periodJson["endDate"];
+        const type = periodJson["studyType"];
+        const degree = periodJson["studyDegree"];
+        const name = `${start} - ${end}, ${convertType(type)}, ${convertDegree(degree)}`
+        setPeriodName(name);
+        setStudyType(type);
+        let details = {};
+        details.startDate = start;
+        details.endDate = end;
+        setPeriodDetails(details);
+      })
+      .catch(e => console.log(e));
+  }, [props.period]);
+
+  useEffect(() => {
+    if (Object.keys(errors).length === 0 && isSubmitting) {
+      submit();
+    } else {
+      console.log(errors);
+    }
+  }, [errors]);
+
+  useEffect(() => {
+    fetch("/api/course-of-study")
+      .then(res => res.json())
+      .then(json => {
+        setCourses(json["courseOfStudies"]);
+      })
+      .catch(e => console.log(e));
+  }, []);
+
+  useEffect(() => {
+    if (props.recruitment === "") return;
+    fetch(`/api/recruitment/${props.recruitment}/rooms`)
+      .then(res => res.json()
```

Comment:
There is no such endpoint on the backend currently

Reply:
It should be added with NIAK-91 though

Reply:
Yes, I used backend from NIAK-91